external/wakaama/examples/client/lwm2mclient.c

```diff
@@ -1185,6 +1185,7 @@ int lwm2m_client_main(int argc, char *argv[])
     if (NULL == objArray[0])
     {
         fprintf(stderr, "Failed to create security object\r\n");
+		close(data.sock);
```

Comment:
It is perhaps a better idea to add a goto statement to near the end of the main function, where data.sock is closed. Furthermore, objects created earlier will have to be freed as well. This needs more detailed checking at the end of the main function, where objects are freed.